bot/wikidata/federation_tester.py

```diff
@@ -46,49 +51,52 @@ def run(self):
+        newtext += '[[:mw:Wikidata Query Service/User Manual/SPARQL Federation endpoints|the manual]] and doing a '
+        newtext += '[https://query.wikidata.org/#SELECT%20%3Fa%20%3Fb%20%3Fc%20WHERE%20%7B%20%0A%20%20SERVICE%20%3Chttp%3A%2F%2Fdata.bibliotheken.nl%2Fsparql%3E%20%7B%0A%20%20%20%20SELECT%20%3Fa%20%3Fb%20%3Fc%20WHERE%20%7B%20%3Fa%20%3Fb%20%3Fc%20%7D%20LIMIT%201%0A%20%20%7D%0A%7D%20LIMIT%201 very simple federated query] ([https://github.com/multichill/toollabs/blob/master/bot/wikidata/federation_tester.py bot source code]).\n\n'
+        newtext += '{| class="wikitable sortable"\n! Link !! Git !! Wiki !! Query works?\n'
         alllist = sorted(set(self.gitlist) | set(self.wikilist))
         for link in alllist:
-            newtext += u'|-\n'
-            newtext += u'| %s\n' % (link,)
+            newtext += '|-\n'
+            newtext += '| %s\n' % (link,)
             if link in self.gitlist:
-                newtext += u'| OK\n' # FIXME: Add pretty icon
+                newtext += '| OK\n' # FIXME: Add pretty icon
             else:
-                newtext += u'| Missing\n'
+                newtext += '| Missing\n'
             if link in self.wikilist:
-                newtext += u'| OK\n' # FIXME: Add pretty icon
+                newtext += '| OK\n' # FIXME: Add pretty icon
             else:
-                newtext += u'| Missing\n'
+                newtext += '| Missing\n'
             queryworks = self.testSparql(link)
             if queryworks:
-                newtext += u'| OK\n' # FIXME: Add pretty icon
+                newtext += '| OK\n' # FIXME: Add pretty icon
             else:
-                newtext += u'| Query failed\n'
-        newtext += u'|}\n[[Category:Wikidata:SPARQL query service]]'
-        print newtext
-        reportpage.put(newtext, summary=u'Updating federation report')
+                newtext += '| Query failed\n'
+        newtext += '|}\n[[Category:Wikidata:SPARQL query service]]'
+        print(newtext)
+        reportpage.put(newtext, summary='Updating federation report')
 
     def testSparql(self, link):
-        query = u"""SELECT ?a ?b ?c WHERE { 
+        query = """SELECT ?a ?b ?c WHERE { 
   SERVICE <%s> {
     SELECT ?a ?b ?c WHERE { ?a ?b ?c } LIMIT 1
   }
 } LIMIT 1""" % (link,)
+        headers = pywikibot.data.sparql.DEFAULT_HEADERS | {
+            'User-Agent': 'federation_tester.py (multichill)',
+        }
         sq = pywikibot.data.sparql.SparqlQuery(max_retries=1, retry_wait=1)
         try:
-            queryresult = sq.select(query)
+            queryresult = sq.select(query, headers=headers)
             if queryresult:
                 return True
-            print queryresult
+            print(queryresult)
             #for resultitem in queryresult:
-            #    qid = resultitem.get('item').replace(u'http://www.wikidata.org/entity/', u'')
+            #    qid = resultitem.get('item').replace('http://www.wikidata.org/entity/', u'')
```

Comment:
only suggestion is rather than fixing this line we could remove the commented section

Reply:
Could probably be removed, yeah (the query doesn’t select `?item` anymore anyway), but I think I’d rather not touch the script more than necessary.

Reply:
Based on the commit data I think this was something I wrote during the Barcelona hackathon in 2018. This job died with the grid. I'll have a look if I can fire it up again.

Reply:
You can just remove this part. That's copy and paste code from something else that got commented out.